### PR TITLE
chore(CHANGELOG): regenerate CHANGELOG with type-scope and KAC template

### LINKED ISSUE #125: bug: current config for project misses key commits in the change log
## Expected Behavior

I expect for Keywords (`BREAKING CHANGE`, etc) and `chore`s to be included in the output of the `make changelog` command.

```
$ make changelog
<a name="unreleased"></a>
## [Unreleased]


<a name="v0.12.0"></a>
## [v0.12.0] - 2021-03-20
### Chore
- bumps version to v0.12.0
- bump golang to 1.16 ([#118](https://github.com/git-chglog/git-chglog/issues/118))
- **ci:** add golangci-lint action and apply linting changes ([#120](https://github.com/git-chglog/git-chglog/issues/120))

### Features
- allow tag sorting by semver ([#124](https://github.com/git-chglog/git-chglog/issues/124))

### BREAKING CHANGE

`JiraIssueId` has been renamed to `JiraIssueID`. This impacts the value for `pattern_maps` in `config.yml`.


[Unreleased]: https://github.com/git-chglog/git-chglog/compare/v0.12.0...HEAD
[v0.12.0]: https://github.com/git-chglog/git-chglog/compare/v0.11.2...v0.12.0
```

## Actual Behavior

```
$ make changelog
<a name="unreleased"></a>
## [Unreleased]


<a name="v0.12.0"></a>
## [v0.12.0] - 2021-03-20
### Features
- allow tag sorting by semver ([#124](https://github.com/git-chglog/git-chglog/issues/124))


[Unreleased]: https://github.com/git-chglog/git-chglog/compare/v0.12.0...HEAD
[v0.12.0]: https://github.com/git-chglog/git-chglog/compare/v0.11.2...v0.12.0
```

## Steps to Reproduce (including precondition)

1. Locally reset repo to commit `9926e07971eb10f03497768655e661ce356b396f`
2. Run `$ make changelog`

## Screenshot on This Problem (if possible)

image

## Your Environment

- OS: Mac OSX
- git-chglog version:  `master` at commit  `9926e07971eb10f03497768655e661ce356b396f`

### PR DESCRIPTION
**Pull request by clok**
## What does this do / why do we need it?

This change makes for a more consistent output of the changlog such that we can achieve an automated tagging and release process for #90. The goal is to get the `make changelog` command to the point that we can run it with the expectations that:

1. The output will include the details that we want from the commits in the log.
1. The output will be idempotent.

This will also address the issues noted in #125. We are using `chore` more often in the repo and those are not currently captured. As such,  a `BREAKING CHANGE` note was missed in our current changelog. I added it in by hand in the previous release.

## How this PR fixes the problem?

This PR adjust the structure of the `CHANGELOG.md` such that it will consider:

1. The `Type`, `Scope` and `Subject` in the header of a commit message.
1. `Types` of `feat`, `fix`, `refactor`, `perf` and `chore`.
1. `Merge` and `Revert` commits.
    - This is definitely up for debate. I kept them in to create a full picture, but I am open to the option of not including them.

Basically, a vanilla `git-chglog --init` run.
generate-new-changelog

## What should your reviewer look out for in this PR?

1. What do you think of the new rules in place for the changelog generation?
2. Do we want to include `Merge` and `Revert` commits?

## Check lists

* [x] Test passed
* [x] Coding style (indentation, etc)

## Which issue(s) does this PR fix?

fixes #125 
related to #90

### CODE REVIEW

**Review by mavogel — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ install:
 
 .PHONY: changelog
 changelog:
-	@git-chglog --next-tag $(tag) $(tag)
+	git-chglog --next-tag $(VERSION) -o CHANGELOG.md
```
Comment (khos2ow) [inline Makefile:26]: To dogfood ourselves, I think it might be better to put `build` as dependency of `changelog` target and calling `./git-chglog` instead of whatever installed somewhere in `$PATH`.

**Review by khos2ow — COMMENTED** [review]
Very minor comment below, otherwise lgtm!

File: Makefile (hunk 1 above)
Comment (clok) [reply Makefile]: Good suggestion. Updates pushed.

I always find the phrase "eat your own dog food" funny. I am too literal ... "But, I don't eat dog food???" 🍔

**Review by khos2ow — APPROVED** [review]
